### PR TITLE
sentieon: update to 201711.04

### PR DESCRIPTION
**Pull request by chapmanb**
Builds off of #9653 to avoid lint and build problems with cb3

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by Lenbok** [comment]
@chapmanb conda isn't showing this as available yet - is package building delayed or something?

**Comment by chapmanb** [comment]
Len -- thanks for checking in on this, I'm not sure what happened here but guess that this somehow got delayed because of the big cb3 build going on. @bioconda/core -- does anyone have a clue how to kick this one to get this latest PR built?

**Comment by rvalieris** [comment]
I am not sure why but, the logs say:
```
BIOCONDA INFO FILTER: not building recipe recipes/sentieon because the same number of builds are in channel(s) and it is not forced.
BIOCONDA INFO Nothing to be done for recipe recipes/sentieon
BIOCONDA INFO BUILD SUMMARY: successfully built 0 of 1 recipes
```

**Comment by chapmanb** [comment]
I tried to bump the Sentieon build number to get this new package built but that still didn't do it. Bummed, I'm not sure what is going on here and would need help from Johannes or Bjorn or someone who understands what is going on with the cb3 build. My guess is that this is in the bulk rebuild somehow but I have no idea why it's not getting done and pushed.

**Comment by mbargull** [comment]
TLDR; remove `or not py27` from the selector for `skip: True` and replace the `python` requirement by `python >=2.7,<3.0a0`.

----
The important build message is
```
BIOCONDA INFO Nothing to be done for recipe recipes/sentieon
```
which means `conda-build 3` didn't pick up a valid configuration for the build.
You can ignore
```
BIOCONDA INFO FILTER: not building recipe recipes/sentieon because the same number of builds are in channel(s) and it is not forced.
```
in this case (fixed that redundant message by https://github.com/bioconda/bioconda-utils/pull/313/files#r200166101).

What happened here is that `conda-build 3` is smarter in selecting the required build matrices. As `python` is only a `run` requirement here, `conda-build` does not build for all different `python` versions since the resulting package isn't affected by `python` at build time (`python` not being in `build`/`host`).
So it'll likely choose `python 3.6.*` currently. But due to `skip: True # [not linux or not py27]`, `[not py27]` evaluates to `True` now so that the build is skipped => `Nothing to be done for recipe recipes/sentieon`.
The gist is that the `[py*]` selectors can be mostly replaced by appropriate version constraints with `conda-build 3`.

**Comment by chapmanb** [comment]
Marcel;
Awesome, thanks so much for the tip. That worked perfectly. Len, the latest build is now available in bioconda and should hopefully be working for you with an update. Let us know if you run into any other problems.

**Comment by Lenbok** [comment]
@chapmanb It's close now! When running sentieon-bwa, I get:

````
/opt/conda/bin/sentieon-bwa: 6: /opt/conda/bin/sentieon-bwa: Bad substitution
````
It looks like the bwa_symlinks.patch has introduced bashisms without also changing the shebang.
